src/python/pants/backend/python/tasks/build_local_python_distributions.py

```diff
@@ -222,25 +250,32 @@ def _prepare_and_create_dist(self, interpreter, shared_libs_product, versioned_t
       # We are including a platform-specific shared lib in this dist, so mark it as such.
       is_platform_specific = True
 
+    versioned_target_fingerprint = versioned_target.cache_key.hash
+
     setup_requires_dir = os.path.join(results_dir, self._SETUP_REQUIRES_SITE_SUBDIR)
     setup_reqs_to_resolve = self._get_setup_requires_to_resolve(dist_target)
     if setup_reqs_to_resolve:
       self.context.log.debug('python_dist target(s) with setup_requires detected. '
                              'Installing setup requirements: {}\n\n'
                              .format([req.key for req in setup_reqs_to_resolve]))
 
-    setup_requires_site_dir = ensure_setup_requires_site_dir(
-      setup_reqs_to_resolve, interpreter, setup_requires_dir, platforms=['current'])
-    if setup_requires_site_dir:
-      self.context.log.debug('Setting PYTHONPATH with setup_requires site directory: {}'
-                             .format(setup_requires_site_dir))
+    setup_reqs_pex_path = os.path.join(
+      setup_requires_dir,
+      'setup-requires-{}.pex'.format(versioned_target_fingerprint))
+    setup_requires_pex = self._build_setup_requires_pex_settings.bootstrap(
+      interpreter, setup_reqs_pex_path,
+      # FIXME: remove this obvious hack! Without this building the dist will fail complaining that
```

Comment:
This should be understood now. The base_requirements of `BuildSetupRequiresPex` already has an un-constrained 'setuptools' requirement.

Reply:
Ah! That makes perfect sense, and let me remove this "obvious hack" in 8d431b20c517f1d13c7cb347c9e17a95fa04780f!